### PR TITLE
Standard formatting of error messages.

### PR DESCRIPTION
**Pull request by audetto**
https://www.gnu.org/prep/standards/html_node/Errors.html

Issue: https://github.com/cc65/cc65/issues/1494

### CODE REVIEW

**Comment by audetto** [comment]
I am still looking if I have missed anything, but the idea is this.

BTW, if you think the string "%s:%u:" should be a macro for future configuration and customisation, it can be done.

File: src/ld65/exports.c [hunk 1]
```diff
@@ -774,17 +774,19 @@ static void PrintUnresolved (ExpCheckFunc F, void* Data)
...
                      "Unresolved external '%s' referenced in:\n",
-                     GetString (E->Name));
+                     name);
             while (Imp) {
                 unsigned J;
                 for (J = 0; J < CollCount (&Imp->RefLines); ++J) {
                     const LineInfo* LI = CollConstAt (&Imp->RefLines, J);
                     fprintf (stderr,
-                         "  %s(%u)\n",
+                         "  %s:%u: Error: Unresolved external '%s'\n",
```
Comment (greg-king5) [inline src/ld65/exports.c:786]: It's redundant.  Maybe you should remove the previous error line, and show only the list of reference lines.

File: src/ld65/exports.c [hunk 2]
```diff
@@ -774,17 +774,19 @@ static void PrintUnresolved (ExpCheckFunc F, void* Data)
         if (E->Expr == 0 && E->ImpCount > 0 && F (E->Name, Data) == 0) {
             /* Unresolved external */
             Import* Imp = E->ImpList;
+            const char * name = GetString (E->Name);
```
Comment (oliverschmidt) [inline src/ld65/exports.c:777]: The predominant style is `char*`, not `char *`.

**Comment by audetto** [comment]
I made the suggested changes, added one more case in `exports.c`.

If one searches `%s(%u)` in the codebase, it still appears in a few places:

1. `_afailed.c`, `mousedemo.c`, `mouse-test.c`: I think these run in the guest os, so not sure they are needed, but could be for uniformity
2. `exports.c`: https://github.com/cc65/cc65/blob/5d84a4ba137e32163390ccd058c32ad169b4ca94/src/ld65/exports.c#L693 I am not too sure what they are
3. `preproc.c`: https://github.com/cc65/cc65/blob/5d84a4ba137e32163390ccd058c32ad169b4ca94/src/cc65/preproc.c#L693 is this a line number?

**Comment by polluks2** [comment]
Yes, number 2 uses GetSourceLine().
No, number 3 is a counter.

**Comment by oliverschmidt** [comment]
1. Please adjust `_afailed.c`. Please leave `mousedemo.c` and `mouse-test.c` alone. They are not about line numbers at all.
2. As @polluks2 wrote.
3. As @polluks2 wrote.

**Pull request merged**